core/vm/contracts.libevm.go

```diff
@@ -122,78 +119,62 @@ type PrecompileEnvironment interface {
 	BlockHeader() (types.Header, error)
 	BlockNumber() *big.Int
 	BlockTime() uint64
-}
-
-//
-// ****** SECURITY ******
-//
-// If you are updating PrecompileEnvironment to provide the ability to call back
-// into another contract, you MUST revisit the evmCallArgs.forceReadOnly flag.
-//
-// It is possible that forceReadOnly is true but evm.interpreter.readOnly is
-// false. This is safe for now, but may not be if recursive calling *from* a
-// precompile is enabled.
-//
-// ****** SECURITY ******
-
-var _ PrecompileEnvironment = (*evmCallArgs)(nil)
```

Comment:
These have all been moved to the new `environment` type in `environment.libevm.go`.